presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractMinMaxAggregationFunction.java

```diff
@@ -261,6 +263,9 @@ private static void compareAndUpdateState(MethodHandle methodHandle, NullableDou
             if ((boolean) methodHandle.invokeExact(value, state.getDouble())) {
                 state.setDouble(value);
             }
+            else if (isNaN(value)) {
```

Comment:
if state became a NaN, we shouldn't be calling the comparison `methodHandle` above, right?